packages/ckeditor5-editor-inline/tests/inlineeditor.js

```diff
@@ -284,6 +286,106 @@ describe( 'InlineEditor', () => {
 				.then( done )
 				.catch( done );
 		} );
+
+		describe( 'configurable editor label (aria-label)', () => {
+			it( 'should be set to the defaut value if not configured', () => {
+				expect( editor.editing.view.getDomRoot().getAttribute( 'aria-label' ) ).to.equal(
+					'Rich Text Editor. Editing area: main'
+				);
+			} );
+
+			it( 'should support the string format', async () => {
+				await editor.destroy();
+
+				editor = await InlineEditor.create( editorElement, {
+					plugins: [ Paragraph, Bold ],
+					label: 'Custom label'
+				} );
+
+				expect( editor.editing.view.getDomRoot().getAttribute( 'aria-label' ) ).to.equal(
+					'Custom label'
+				);
+			} );
+
+			it( 'should support object format', async () => {
+				await editor.destroy();
+
+				editor = await InlineEditor.create( editorElement, {
+					plugins: [ Paragraph, Bold ],
+					label: {
+						main: 'Custom label'
+					}
+				} );
+
+				expect( editor.editing.view.getDomRoot().getAttribute( 'aria-label' ) ).to.equal(
+					'Custom label'
+				);
+			} );
+
+			it( 'should keep an existing value from the source DOM element', async () => {
+				await editor.destroy();
+
+				editorElement.setAttribute( 'aria-label', 'Pre-existing value' );
+				editor = await InlineEditor.create( editorElement, {
+					plugins: [ Paragraph, Bold ]
+				} );
+
+				expect( editor.editing.view.getDomRoot().getAttribute( 'aria-label' ), 'Keep value' ).to.equal(
+					'Pre-existing value'
+				);
+
+				await editor.destroy();
+
+				expect( editorElement.getAttribute( 'aria-label' ), 'Restore value' ).to.equal( 'Pre-existing value' );
+			} );
+
+			it( 'should override the existing value from the source DOM element', async () => {
+				await editor.destroy();
+
+				editorElement.setAttribute( 'aria-label', 'Pre-existing value' );
+				editor = await InlineEditor.create( editorElement, {
+					plugins: [ Paragraph, Bold ],
+					label: 'Custom label'
+				} );
+
+				expect( editor.editing.view.getDomRoot().getAttribute( 'aria-label' ), 'Override value' ).to.equal(
+					'Custom label'
+				);
+
+				await editor.destroy();
+
+				expect( editorElement.getAttribute( 'aria-label' ), 'Restore value' ).to.equal( 'Pre-existing value' );
```

Comment:
More general question here, but regarding this behavior (restoring initial `aria-label` attribute value on editor destroy) - how this behaves if there is no `aria-label` set initially? Is it back to empty after editor destroy or `aria-label` is kept? (not sure how it should behave TBH, just food for though).

Reply:
It should work out just fine. See how the attributes are restored upon `(Editing)View#detachDomRoot()`:

https://github.com/ckeditor/ckeditor5/blob/54efa9938f46463d11f450b009b63140fe9bf946/packages/ckeditor5-engine/src/view/view.ts#L336-L343

First, we nuke all existing ones and then restore the cached values [from when the root was attached](https://github.com/ckeditor/ckeditor5/blob/54efa9938f46463d11f450b009b63140fe9bf946/packages/ckeditor5-engine/src/view/view.ts#L263-L291). This way we guarantee nothing from the editor's lifecycle will ever be able to stay (leak).

Reply:
Ok, that's good 👍 Thanks for the code references and explanation 👍